tools/c7n_azure/c7n_azure/query.py

```diff
@@ -48,5 +48,30 @@ def __new__(cls, name, parents, attrs):
 
 
 @six.add_metaclass(QueryMeta)
-class QueryResourceManager(object):
-    pass
+class QueryResourceManager(ResourceManager):
+
+    def __init__(self, data, options):
+        super(QueryResourceManager, self).__init__(data, options)
+        self.source = self.get_source(self.source_type)
+
+    def get_permissions(self):
+        return ()
+
+    def get_source(self, source_type):
+        return sources.get(source_type)(self)
+
+    def get_cache_key(self, query):
+        return {'source_type': self.source_type, 'query': query}
+
+    @property
+    def source_type(self):
+        return self.data.get('source', 'describe-azure')
```

Comment:
You need to add the following or get_sources(...) will throw since 'describe-azure' is never registered with the PluginRegistry.

```
@sources.register('describe-azure')
class DescribeSource(object):

    def __init__(self, manager):
        self.manager = manager
        self.query = ResourceQuery(manager.session_factory)

    def get_resources(self, query):
        return self.query.filter(self.manager)

    def get_permissions(self):
        return ()

    def augment(self, resources):
        return resources
```